components/ParameterForm.vue

```diff
@@ -76,28 +76,25 @@
         <CIcon v-else icon="cilArrowRight" />
       </CButton>
     </CForm>
-    <CAlert v-else-if="props.metadataFetchStatus === 'error'" color="warning">
-      Failed to retrieve metadata from R API. {{ metadataFetchError }}
+    <CAlert v-else-if="appStore.metadataFetchStatus === 'error'" color="warning">
+      Failed to retrieve metadata from R API. {{ appStore.metadataFetchError }}
     </CAlert>
-    <CSpinner v-else-if="props.metadataFetchStatus === 'pending'" />
+    <CSpinner v-else-if="appStore.metadataFetchStatus === 'pending'" />
```

Comment:
Does this ever get shown? It looks to me like the metadataFetchStatus doesn't get set in the store until the fetch is completed, at which point presumbaly it's no longer pending. The approach we normally take would be to set a "fetching" flag or similar in the store before we start the request, then unset it after request completes. 
Could also just show the spinner if the metadata isn't set yet. and there isn't an error..

Reply:
Good catch, I guess fetch status can never be 'pending' any more.
I have put it the second approach you describe.

Reply:
... in a different PR, of course https://github.com/jameel-institute/daedalus-web-app/pull/28/commits